contracts/VotingAdapter.vy

```diff
@@ -0,0 +1,224 @@
+    amount: uint256
+
+
+event ETHRecovered:
+    amount: uint256
+
+
+event OwnerChanged:
+    owner: address
+
+
+ZERO_BYTES32: constant(
+    bytes32
+) = 0x0000000000000000000000000000000000000000000000000000000000000000
+
+VOTING_CONTRACT_ADDR: immutable(address)
+SNAPSHOT_DELEGATE_CONTRACT_ADDR: immutable(address)
+DELEGATION_CONTRACT_ADDR: immutable(address)
+
+owner: public(address)
+
+
+@external
+def __init__(
+    voting_addr: address,
+    snapshot_delegate_addr: address,
+    delegation_addr: address,
+    owner: address,
+):
+    """
+    @notice Initialize source contract implementation.
+    @param voting_addr Address of the Voting contract
+    @param snapshot_delegate_addr Address of the Shapshot Delegate contract
+    @param delegation_addr Address of the voting power delegation contract
+    @param owner Address to recover tokens and ether to
+    """
+    assert owner != empty(address), "zero owner"
+    self.owner = owner
+    VOTING_CONTRACT_ADDR = voting_addr
+    SNAPSHOT_DELEGATE_CONTRACT_ADDR = snapshot_delegate_addr
+    DELEGATION_CONTRACT_ADDR = delegation_addr
+
+
+@external
+@view
+def encode_aragon_vote_calldata(voteId: uint256, supports: bool) -> Bytes[1000]:
+    """
+    @notice Encode calldata for use in VestingEscrow
+    @param voteId Id of the vote
+    @param supports Support flag true - yea, false - nay
+    """
+    return _abi_encode(voteId, supports)
+
+
+@external
+def aragon_vote(abi_encoded_params: Bytes[1000]):
+    """
+    @notice Cast vote on Aragon voting
+    @param abi_encoded_params Abi encoded data for call. Can be obtained from encode_aragon_vote_calldata
+    """
+    vote_id: uint256 = empty(uint256)
+    supports: bool = empty(bool)
+    vote_id, supports = _abi_decode (abi_encoded_params, (uint256, bool))
+    IVoting(VOTING_CONTRACT_ADDR).vote(
+        vote_id, supports, False
+    )  # dev: third arg is deprecated
+
+
+@external
+@view
+def encode_snapshot_set_delegate_calldata(delegate: address) -> Bytes[1000]:
+    """
+    @notice Encode calldata for use in VestingEscrow
+    @param delegate Address of the delegate
+    """
+    return _abi_encode(delegate)
+
+
+@external
+def snapshot_set_delegate(abi_encoded_params: Bytes[1000]):
+    """
+    @notice Delegate Snapshot voting power of all available tokens
+    @param abi_encoded_params Abi encoded data for call. Can be obtained from encode_snapshot_set_delegate_calldata
+    """
+    delegate: address = empty(address)
+    delegate = _abi_decode (abi_encoded_params, (address))
+    IDelegation(SNAPSHOT_DELEGATE_CONTRACT_ADDR).setDelegate(
+        ZERO_BYTES32, delegate
+    )  # dev: null id allows voting at any snapshot space
+
+
+@external
+@view
+def encode_delegate_calldata(delegate: address) -> Bytes[1000]:
+    """
+    @notice Encode calldata for use in VestingEscrow
+    @param delegate Address of the delegate
+    """
+    return _abi_encode(delegate)
+
+
+@external
+def delegate(abi_encoded_params: Bytes[1000]):
```

Comment:
Maybe also add a prefix to the method? Something like `aragon_delegate` or similar to snapshot delegation `aragon_set_delegate`.

Reply:
We have no delegation implementation ATM. And I'm not 100% sure that it will be only aragon delegate